extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java

```diff
@@ -137,8 +142,26 @@ public void nativeZip(OutputTargetBuildItem target,
                 }
             }
             addZipEntry(zip, nativeImage.getPath(), executableName, 0755);
+
+            final GraalVM.Version graalVMVersion = nativeImageRunner.getBuildRunner().getGraalVMVersion();
+            if (graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) >= 0) {
+                // See https://github.com/oracle/graal/issues/4921
+                try (DirectoryStream<Path> sharedLibs = Files.newDirectoryStream(nativeImage.getPath().getParent(),
+                        "*.{so,dll}")) {
```

Comment:
I know the use-case if for Amazon Lambda, but should this also include `jnilib` to cover macOS?
CC @Karm

Reply:
I have no access to any mac device, so I don't know. @galderz

Reply:
AFAIK `dylib` is the extension on macos.

Reply:
ah, system based libraries, not the ones in JAR files.

Anyway, should be `dylib` added?

Reply:
> Anyway, should be dylib added?

I guess if this functions can run on macOS then yeah, but don't know how feasible that is.